### PR TITLE
[Core] Remove template argument from Node

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This PR removes template argument (both dimension and DoF). @pooyan-dadvand ask me to put as reviewers @rubenzorrilla and @roigcarlo 

Many files involved:

1. Remove template from `node.h` (only file I suggest to look in depth, changes reduced to minimum)
2. Redefine declarations

**TODO**: 
- [ ] Part of the code should be moved to source file to reduce the compilation times. (future PR after speaking with @pooyan-dadvand)
- [ ] We should also at least explicitly instantiate `Dof<double>` in order to reduce compilation time

**🆕 Changelog**

- Remove dimension template argument
- Remove DoF template argument
- Replace definitions

### CODE REVIEW

File: applications/CableNetApplication/custom_processes/apply_weak_sliding_process.h
Comment (roigcarlo) [inline applications/CableNetApplication/custom_processes/apply_weak_sliding_process.h:45]: I know its a little bit picky, but change this to `Node::Pointer` or directly use them, since there is no longer template it does not have much sense to have an alias here (same as with `NodeType`).

File: applications/CSharpWrapperApplication/custom_includes/kratos_wrapper.h
Comment (roigcarlo) [inline applications/CSharpWrapperApplication/custom_includes/kratos_wrapper.h:30]: We should change those and directly use `Node`

File: applications/DEMApplication/custom_elements/spheric_particle.h
Comment (roigcarlo) [inline applications/DEMApplication/custom_elements/spheric_particle.h:108]: Not related, but this is dangerous, we should take a look....

File: kratos/utilities/qef_utility.h
Comment (roigcarlo) [inline kratos/utilities/qef_utility.h:63]: Same as the others but for this syntax as well

File: applications/FluidDynamicsApplication/custom_elements/alternative_d_vms_dem_coupled.h
Comment (roigcarlo) [inline applications/FluidDynamicsApplication/custom_elements/alternative_d_vms_dem_coupled.h:61]: All this comments should be gone now

File: applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp
Comment (roigcarlo) [inline applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp:453]: :+1:

File: applications/GeoMechanicsApplication/custom_processes/apply_constant_interpolate_line_pressure_process.hpp
Comment (roigcarlo) [inline applications/GeoMechanicsApplication/custom_processes/apply_constant_interpolate_line_pressure_process.hpp:172]: Probably we should check those as well...

**Review by roigcarlo — COMMENTED** [review]
I will be super annoying with details because I don't want to review another such PR.

It will come by batches, please don't make any change until I finish so I can keep track of what I've looked at:

- [x] CSharp
- [x] CableNet
- [x] Chimera
- [x] Cosim
- [x] CompressiblePotentialFlow
- [x] ConstitutiveLaws
- [x] ConstitutiveModels
- [x] ContactMechanics
- [x] ContactStructuralMechanis
- [x] ConvectionDiffusion
- [x] DEM
- [x] Dam
- [x] DelaunayMeshing
- [x] ExaquteSandox
- [x] FSI
- [x] FemToDem
- [x] FluidDynamics
- [x] FluidDynamocsBiomedical
- [x] FluidTransport
- [x] FreeSurface
- [x] GeoMechanics
- [ ] ....
- [ ] Core

File: applications/CableNetApplication/custom_processes/apply_weak_sliding_process.h
Comment (loumalouomega) [reply applications/CableNetApplication/custom_processes/apply_weak_sliding_process.h:45]: As you said to not commit anything until you are done I will create a branch from this branch an generate an independent PR, are you Okay with that?

Comment (loumalouomega) [reply applications/CableNetApplication/custom_processes/apply_weak_sliding_process.h:45]: Anycase what you are asking me in this is making even more complex to look the changes because it will imply changing a lot of things, better to do in a fiture PR.

File: applications/DEMApplication/custom_elements/spheric_particle.h
Comment (loumalouomega) [reply applications/DEMApplication/custom_elements/spheric_particle.h:108]: @KratosMultiphysics/dem

File: applications/CSharpWrapperApplication/custom_includes/kratos_wrapper.h
Comment (loumalouomega) [reply applications/CSharpWrapperApplication/custom_includes/kratos_wrapper.h:30]: I know, but those changes will make everything more difficult to review

File: applications/GeoMechanicsApplication/custom_processes/apply_constant_interpolate_line_pressure_process.hpp
Comment (loumalouomega) [reply applications/GeoMechanicsApplication/custom_processes/apply_constant_interpolate_line_pressure_process.hpp:172]: @anoordam This pongs all geomechanics team

File: applications/DEMApplication/custom_elements/spheric_particle.h
Comment (maceligueta) [reply applications/DEMApplication/custom_elements/spheric_particle.h:108]: I see no real problem. It’s part of a memory buffer, so it must be fast and parallel and avoid copies of smart pointers.

**Comment by roigcarlo** [comment]
OK :+1:

**Comment by philbucher** [comment]
@loumalouomega after modifying every single line of code in Kratos 😂

image

**Comment by loumalouomega** [comment]
> @loumalouomega after modifying every single line of code in Kratos 😂
>

**Comment by philbucher** [comment]
Hardly ever have I seen such a perfect reply 😂😂😂

**Review by roigcarlo — COMMENTED** [review]
@loumalouomega We will meet tomorrow, so give me a day more before approving, I am very afraid to to give green light without having spoke with the rest of the TC

**Comment by roigcarlo** [comment]
1 - @KratosMultiphysics/technical-committee agrees with the change. Pending on @pooyan-dadvand approval as well.
2 - Before proceding with the merge, we will create a pinged issue with all ping to let everyonw know about the change
3 - We will accept and merge in one week time.

**Comment by loumalouomega** [comment]
> 1 - @KratosMultiphysics/technical-committee agrees with the change. Pending on @pooyan-dadvand approval as well. 2 - Before proceding with the merge, we will create a pinged issue with all ping to let everyonw know about the change 3 - We will accept and merge in one week time.

image

**Comment by loumalouomega** [comment]
> > 1 - @KratosMultiphysics/technical-committee agrees with the change. Pending on @pooyan-dadvand approval as well. 2 - Before proceding with the merge, we will create a pinged issue with all ping to let everyonw know about the change 3 - We will accept and merge in one week time.
> 
> image

Soon

**Comment by jginternational** [comment]
image

**Comment by loumalouomega** [comment]
Today is the date

**Comment by loumalouomega** [comment]
> Today is the date

TODAY!

image

**Comment by loumalouomega** [comment]
CI passed @roigcarlo

**Review by roigcarlo — APPROVED** [review]
:pray:

**Pull request merged**
